### PR TITLE
update prometheus no client golang at grafana/mimir-prometheus@fdf902d

### PR DESCRIPTION
**Pull request by krajorama**
#### What this PR does

Update mimir-prometheus at https://github.com/grafana/mimir-prometheus/commit/fdf902dd68d9ae0da99356ce6729d052f867a926

The remote read client interface has changed in https://github.com/prometheus/prometheus/pull/11379
I've updated the users and also added native histogram support to backfill.

Keeps client_golang version as it has some bugs.
https://github.com/prometheus/client_golang/issues/1605
https://github.com/prometheus/client_golang/issues/1607

### CODE REVIEW

**Review by charleskorn — APPROVED** [review]

**Review by pstibrany — APPROVED** [review]
Mimir code changes lgtm. I haven't looked closely at Prom code changes.

File: pkg/ruler/remotequerier.go [hunk 1]
```diff
@@ -209,7 +222,9 @@ func (q *RemoteQuerier) Read(ctx context.Context, query *prompb.Query) (*prompb.
 	if len(rdResp.Results) != 1 {
 		return nil, errors.Errorf("responses: want %d, got %d", 1, len(rdResp.Results))
 	}
-	return rdResp.Results[0], nil
+
+	res := rdResp.Results[0]
+	return remote.FromQueryResult(sortSeries, res), nil
```
Comment (dimitarvdimitrov) [inline pkg/ruler/remotequerier.go:227]: I don't understand why `sortSeries` is a parameter now. Did the remote-read API also change? Previously they were supposed to be returned sorted by the server. Are we supposed to pass `sortSeries` to the server?

**Review by dimitarvdimitrov — COMMENTED** [review]
had a look at pkg/querier, ruler, and mimirtool. pkg/distributor seems copied from upstream and I hope someone reviewed it there.

File: pkg/ruler/remotequerier.go [hunk 2]
```diff
@@ -147,7 +149,7 @@ func NewRemoteQuerier(
 
 // Read satisfies Prometheus remote.ReadClient.
 // See: https://github.com/prometheus/prometheus/blob/1291ec71851a7383de30b089f456fdb6202d037a/storage/remote/client.go#L264
-func (q *RemoteQuerier) Read(ctx context.Context, query *prompb.Query) (*prompb.QueryResult, error) {
+func (q *RemoteQuerier) Read(ctx context.Context, query *prompb.Query, sortSeries bool) (storage.SeriesSet, error) {
```
[the remark this answers is not in the document]
Comment (krajorama) [reply pkg/ruler/remotequerier.go:152]: mine doesn't , updating

File: pkg/ruler/remotequerier.go (hunk 1 above)
Comment (dimitarvdimitrov) [reply pkg/ruler/remotequerier.go:227]: strange

**Review by dimitarvdimitrov — APPROVED** [review]
LGTM, thanks

**Pull request merged**